src/worker/plugins/run.ts

```diff
@@ -58,11 +75,20 @@ export async function runPluginsOnBatch(server: PluginsServer, batch: PluginEven
                     await processError(server, pluginConfig, error, returnedEvents[0])
                     server.statsd?.increment(`plugin.${pluginConfig.plugin?.name}.process_event_batch.ERROR`)
                 }
+
+                const timeSpentRunning = new Date().getTime() - timer.getTime()
                 server.statsd?.timing(`plugin.${pluginConfig.plugin?.name}.process_event_batch`, timer)
                 server.statsd?.timing('plugin.process_event_batch', timer, 0.2, {
                     plugin: pluginConfig.plugin?.name ?? '?',
                     teamId: teamId.toString(),
                 })
+                posthog.capture('$plugin_running_duration', {
+                    team: teamId,
+                    plugin: pluginConfig.plugin?.name,
+                    time_ms: timeSpentRunning,
+                    time_seconds: Math.round(timeSpentRunning / 10) / 1000,
+                    function: 'processEventBatch',
+                })
```

Comment:
same feedback as above, yet also having a 7-line code block like this identical to one a few lines above (and one a few lines below) screams DRY and is another smell that something can be consolidated.